Dockerfile

```diff
@@ -1,33 +1,48 @@
-FROM python:3.6-alpine3.10 as base
+FROM python:3.7-alpine3.12 as base
 
 # Copy the requirements & code and install them
 # Do this in a separate image in a separate directory
 # to not have all the build stuff in the final image
 FROM base AS builder
-RUN apk update
 # Needed to build cffi
-RUN apk add python-dev build-base libffi-dev
+RUN apk add --no-cache \
+        python3 \
+    && apk add --no-cache --virtual .build-deps \
+        git \
+        gcc \
+        build-base \
+        libffi-dev \
+    && mkdir /stuff && cd /stuff || exit 1 \
+    && git clone https://github.com/ncopa/su-exec.git su-exec-clone \
+    && cd su-exec-clone || exit 1 \
```

Comment:
I'd prefer we use [gosu](https://github.com/tianon/gosu) instead of `su-exec`. I recognize it's larger, but they publish built binaries for their releases, which we could use rather than pulling in all the build machinery, cloning, making, etc. I'm also personally more familiar with `gosu`, and it seems to be the more commonly used of the two (also easier for most of us to introspect go dependencies than C)

Reply:
In general I agree, but this would also mean to:
* Install `Go` into the docker image or
* Use a `Go` based docker image and install `python`. 

This also means:
* The docker image gets bigger.
* There will be an additional system dependency.

This is basically why I decided against `gosu` in the first place.

Just a few notes to the current implementation (coming with this PR):
* The runtime docker image is based on the `base` image, which does not contain the build dependencies - it is pure `python:3.8-alpine3.12`.
* The `builder` image (which also builds `su-exec`) is cleaned at the very end and the build dependencies are removed as well.

If you like, I can still implement `gosu`.

Reply:
Do you need go to run a go binary? I thought they could run without golang installed on the system. `gosu` publishes binary releases, so I figured we might be able to use that.

If that's not the case, no need to bother with `gosu`!

Reply:
I didn't know - I tested it and it works without installing `Go` itself. Great. Changed :+1: 

Some changes:
* I replaced `su-exec` with `gosu`.
* I added the signature check (for `gosu`) according to the recommendation from https://github.com/tianon/gosu/blob/master/INSTALL.md.
* I restructured the docker layers to separate the single build stages and their system dependencies. Also this separates the data within those intermediary docker images, even makes them smaller. - This only affects the docker image build.

Could you please have another look @mplanchard ?

Thanks a lot.